test-framework/test-suites/integration/tests/report/test_report_host_bootfile.py

```diff
@@ -0,0 +1,145 @@
+		assert result.rc == 0
+		result_specific = host.run('stack report host bootfile backend-1-1')
+		assert result_specific.rc == 0
+		assert result.stdout == result_specific.stdout
+		assert result.stdout == ""
+
+		result = host.run('stack set host boot backend-1-1 action=install')
+		assert result.rc == 0
+		result = host.run('stack report host bootfile')
+		assert result.rc == 0
+		result_specific = host.run('stack report host bootfile backend-1-1')
+		assert result_specific.rc == 0
+		assert result.stdout == result_specific.stdout
+		assert result.stdout == ""
+
+
+	def test_report_host_bootfile_for_host_with_interface_but_no_IP_or_network(self, host):
+		result = host.run('stack add host backend-1-1')
+		assert result.rc == 0
+		result = host.run('stack add host interface backend-1-1 interface=eth1 default=True mac=52:54:00:6e:50:91')
+		assert result.rc == 0
+
+		result = host.run('stack report host bootfile')
+		assert result.rc == 0
+		result_specific = host.run('stack report host bootfile backend-1-1')
+		assert result_specific.rc == 0
+		assert result.stdout == result_specific.stdout
+		assert result.stdout == ""
+
+		result = host.run('stack set host boot backend-1-1 action=os')
+		assert result.rc == 0
+		result = host.run('stack report host bootfile')
+		assert result.rc == 0
+		result_specific = host.run('stack report host bootfile backend-1-1')
+		assert result_specific.rc == 0
+		assert result.stdout == result_specific.stdout
+		assert result.stdout == ""
+
+		result = host.run('stack set host boot backend-1-1 action=install')
+		assert result.rc == 0
+		result = host.run('stack report host bootfile')
+		assert result.rc == 0
+		result_specific = host.run('stack report host bootfile backend-1-1')
+		assert result_specific.rc == 0
+		assert result.stdout == result_specific.stdout
+		assert result.stdout == ""
+
+
+	def test_report_host_bootfile_for_combination_of_no_host_interface_and_a_host_with_full_config(self, host):
+		result = host.run('stack add host backend-2-2')
+		assert result.rc == 0
+		result = host.run('stack set host boot backend-2-2 action=os')
+		assert result.rc == 0
+
+		result = host.run('stack add host backend-1-1')
+		assert result.rc == 0
+		result = host.run('stack add network public address=192.172.0.0 mask=255.255.255.0 pxe=True')
+		assert result.rc == 0
+		result = host.run('stack add host interface backend-1-1 interface=eth1 ip=192.172.0.2 network=public default=True mac=52:54:00:6e:50:91')
+		assert result.rc == 0
+		result = host.run('stack set host boot backend-1-1 action=os')
+		assert result.rc == 0
+
+		result = host.run('stack report host bootfile')
+		assert result.rc == 0
+		result_specific = host.run('stack report host bootfile backend-1-1')
+		assert result_specific.rc == 0
+		assert result.stdout == result_specific.stdout
+
+
+	def test_report_host_bootfile_for_combination_of_host_with_interface_and_no_IP_or_network_and_a_host_with_full_config(self, host):
```

Comment:
114 characters might be too long for even a test function name :)

Maybe move some of that context to a comment inside the test